tasks/Task.cpp

```diff
@@ -13,37 +18,638 @@ Task::~Task()
+        GenApi::CFloatPtr exposure_time = m_device->GetNodeMap()->GetNode("ExposureTime");
+
+        auto max_exposure_time = exposure_time->GetMax();
+        auto min_exposure_time = exposure_time->GetMin();
+        auto setpoint = _exposure_time.get().toMicroseconds();
+
+        if (setpoint > max_exposure_time) {
+            LOG_WARN_S << "Exposure time exceeds maximum value. Setting Exposure to "
+                          "maximum value: "
+                       << max_exposure_time << "us" << endl;
+            setpoint = max_exposure_time;
+        }
+        else if (setpoint < min_exposure_time) {
+            LOG_WARN_S << "Exposure time exceeds minimum value. Setting Exposure to "
+                          "minimum value: "
+                       << min_exposure_time << "us" << endl;
+            setpoint = min_exposure_time;
+        }
+        exposure_time->SetValue(setpoint);
+
+        if (auto current = exposure_time->GetValue() != setpoint) {
+            LOG_WARN_S << "Exposure Time value differs from setpoint: " << setpoint
+                       << " current: " << current << endl;
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool Task::acquisitionConfiguration()
+{
+    LOG_INFO_S << "Configuring Acquisition." << endl;
+
+    LOG_INFO_S << "Setting Acquisition Mode." << endl;
+    Arena::SetNodeValue<GenICam::gcstring>(m_device->GetNodeMap(),
+        "AcquisitionMode",
+        "Continuous");
+
+    LOG_INFO_S << "Setting Acquisition Frame Rate." << endl;
+    // Enable Acquisition Frame Rate
+    Arena::SetNodeValue<bool>(m_device->GetNodeMap(), "AcquisitionFrameRateEnable", true);
+
+    // get Acquisition Frame Rate node
+    GenApi::CFloatPtr acquisition_frame_rate =
+        m_device->GetNodeMap()->GetNode("AcquisitionFrameRate");
+
+    GenApi::CFloatPtr pExposureTime = m_device->GetNodeMap()->GetNode("ExposureTime");
+    LOG_INFO_S << "Exposure Time " << pExposureTime->GetValue() << " milliseconds"
+                << std::endl;
+
+    acquisition_frame_rate->SetValue(_frame_rate.get());
+
+    if (auto current = Arena::GetNodeValue<double>(m_device->GetNodeMap(),
+                           "AcquisitionFrameRate") != _frame_rate.get()) {
+        LOG_ERROR_S << "Frame rate was not set correctly. Setpoint: " << _frame_rate.get()
+                    << ". Current: " << current << endl;
+        return false;
+    }
+
+    return true;
+}
+
+bool Task::factoryReset()
+{
+    GenApi::CCommandPtr trigger_software =
+        m_device->GetNodeMap()->GetNode("DeviceFactoryReset");
+    trigger_software->Execute();
+    bool online = false;
+    while (!online) {
+        try {
+            GenApi::CIntegerPtr width = m_device->GetNodeMap()->GetNode("Width");
+            width->GetMax();
+            online = true;
+        }
+        catch (GenICam::TimeoutException& ex) {
+            usleep(500000);
+        }
+    }
```

Comment:
Why not?

```
while (true) {
    try {
         ...
         break;
    }
    catch (...) {
        ...
    }
}
```

Reply:
I just want to catch when there is a timeout, (so it won't keep triggering a lot while the camera is restarting)

Reply:
Your code and @jhonasiv's proposal look functionally the same. Two different ways to express the same code ...

Anyways, I'm more concerned by the lack of an actual timeout ... what if we never manage contact ?

Reply:
I'll add a timeout param for it.